week3/homework/HackYourTemperature/server.js

```diff
@@ -0,0 +1,34 @@
+const express = require("express");
+const exphbs = require("express-handlebars");
+const fetch = require("node-fetch");
+
+const app = express();
+app.set("view engine", "handlebars");
+app.engine("handlebars", exphbs({ defaultLayout: false }));
+app.use(express.json());
+app.use(express.urlencoded({ extended: true }));
+
+app.get("/", (req, res) => {
+  res.render("index");
+});
+
+app.post("/weather", (req, res) => {
+  const cityName1 = req.body.cityName;
+  const API_KEY = require("./sources/keys.json").API_KEY;
```

Comment:
Not sure why would you need a require in this function, usually this would be **only** if you have a changing value, that is being constantly changing with time, so you get latest value.
but in you case you don't, so placing it on the top of the file would be better as for reading what is needed for the file, and the 'API_KEY' is needed.

on other thing I would love to have more often is using more features of javsacript such as destructuring:
 ```suggestion
  const { API_KEY } = require("./sources/keys.json");
```

It's there, why not using it?
here and someother place as well ;)